File: programming 1/lab 6/src/main/java/server/commands/SaveCommand.java
package server.commands;

import common.communication.Response;
import server.managers.CollectionManager;

/**
 * Команда для сохранения коллекции в файл.
 */
public class SaveCommand extends Command {
    private final CollectionManager cm;
    public SaveCommand(CollectionManager cm) {
        super("save", "сохранить коллекцию в файл");
        this.cm = cm;
    }

    /**
     * Выполняет команду без аргументов.
     * @param arg аргумент команды (в данной команде не используется).
     */
    @Override
    public Response execute(String arg) {
        if (arg != null) throw new IllegalArgumentException(String.format("Команда %s не принимает аргументы\n", getName()));
        else cm.save();
        return null;
    }
}

File: programming 1/lab 6/kotlin/lab 6/src/main/java/server/commands/SaveCommand.kt
package server.commands

import common.communication.Response
import common.communication.ResponseStatus
import server.managers.CollectionManager

/**
 * Команда для сохранения коллекции в файл.
 */
class SaveCommand(private val cm: CollectionManager) : Command("save", "сохранить коллекцию в файл") {
    /**
     * Выполняет команду без аргументов.
     * @param arg аргумент команды (в данной команде не используется).
     */
    override fun execute(arg: String?): Response {
        require(arg == null) { "Команда $name не принимает аргументы" }
        return cm.save()
    }
}
